src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -253,21 +253,126 @@ public UnspentCoinModel[] ListUnspent(int minConfirmations = 1, int maxConfirmat
                             Confirmations = spendableTx.Confirmations,
                             IsSpendable = spendableTx.Transaction.IsSpendable(),
                             IsSolvable = spendableTx.Transaction.IsSpendable() // If it's spendable we assume it's solvable.
-                            });
+                        });
                     }
                 }
             }
 
             return unspentCoins.ToArray();
         }
 
+        [ActionName("sendmany")]
+        [ActionDescription("Creates and broadcasts a transaction which sends outputs to multiple addresses.")]
+        public async Task<uint256> SendManyAsync(string fromAccount, string addressesJson, int minConf = 1, string comment = null, string subtractFeeFromJson = null, bool isReplaceable = false, int? confTarget = null, string estimateMode = "UNSET")
+        {
+            if (string.IsNullOrEmpty(addressesJson))
+                throw new RPCServerException(RPCErrorCode.RPC_INVALID_PARAMETER, "No valid output addresses specified.");
+
+            var addresses = new Dictionary<string, decimal>();
+            try
+            {
+                // Outputs addresses are keyvalue pairs of address, amount. Translate to Receipient list.
+                addresses = JsonConvert.DeserializeObject<Dictionary<string, decimal>>(addressesJson);
+            }
+            catch (JsonSerializationException ex)
+            {
+                throw new RPCServerException(RPCErrorCode.RPC_PARSE_ERROR, ex.Message);
+            }
+
+            if (addresses.Count == 0)
+                throw new RPCServerException(RPCErrorCode.RPC_INVALID_PARAMETER, "No valid output addresses specified.");
+
+            // Optional list of addresses to subtract fees from.
+            IEnumerable<BitcoinAddress> subtractFeeFromAddresses = null;
+            if (!string.IsNullOrEmpty(subtractFeeFromJson))
+            {
+                try
+                {
+                    subtractFeeFromAddresses = JsonConvert.DeserializeObject<List<string>>(subtractFeeFromJson).Select(i => BitcoinAddress.Create(i, this.fullNode.Network));
+                }
+                catch (JsonSerializationException ex)
+                {
+                    throw new RPCServerException(RPCErrorCode.RPC_PARSE_ERROR, ex.Message);
+                }
+            }
+
+            var recipients = new List<Recipient>();
+            foreach (var address in addresses)
+            {
+                // Check for duplicate recipients
+                var recipientAddress = BitcoinAddress.Create(address.Key, this.fullNode.Network).ScriptPubKey;
+                if (recipients.Any(r => r.ScriptPubKey == recipientAddress))
+                    throw new RPCServerException(RPCErrorCode.RPC_INVALID_PARAMETER, string.Format("Invalid parameter, duplicated address: {0}.", recipientAddress));
+
+                var amount = Money.Coins(address.Value);
+                if (amount <= 0)
```

Comment:
This is checked when the transaction is built and ArgumentOutOfRange exception is thrown by wallet. Could potentially replace this check with a catch block for this exception.

Reply:
Ok yeah I have now also removed this check 👍